lib/llvm-backend/src/code.rs

```diff
@@ -6054,7 +6395,7 @@ impl FunctionCodeGenerator<CodegenError> for LLVMFunctionCodeGenerator {
                     Some(0),
                 );
                 let old = builder.build_int_z_extend(old, intrinsics.i64_ty, &state.var_name());
-                state.push1(old);
+                state.push1_extra(old, ExtraInfo::arithmetic_f64());
```

Comment:
`Operator::I32AtomicRmwAdd` is different than the sub variant here, the sub has `arithmetic_f32`